pkgdb2client/admin.py

```diff
@@ -391,6 +390,35 @@ def __handle_request_branch(actionid, action):
                 'status': 'Approved'
             }
         )
+        return data
+
+    msgs = utils.check_branch_creation(
+        PKGDBCLIENT,
+        action['package']['name'],
+        action['collection']['branchname'],
+        action['user']
+    )
+
+    if package is not None:
+        branch = action['info']['pkg_collection']
+
+        # Requests for Fedora branches can be automatically approved if they
+        # are from a package admin. Usually package DB approves them directly
+        # only when the package was created at the same time the branches were
+        # requested, it needs to be approved by a pkgdb admin.
+        if branch[0] == "f":
+            for acl in package["acls"]:
+                if acl["fas_name"] == action["user"] and \
+                        acl["acl"] == "approveacls" and \
+                        acl["status"] == "Approved":
+                    if len(msgs["bad"]) == 0:
+                        print("Auto approving Fedora Branch request for "
+                              "package admin {0}".format(action["user"]))
+                        return approve()
```

Comment:
Just styling, but I think I would have approached it by setting ``decision`` to ``approve`` and call ``_ask_what_to_do`` only if it's not None and then left the rest of the logic un-changed (ie: not have the ``approve`` inner method, which I find a little harder to read).